pydantic/main.py

```diff
@@ -337,7 +337,7 @@ def __init__(__pydantic_self__, **data: Any) -> None:
         object_setattr(__pydantic_self__, '__fields_set__', fields_set)
         __pydantic_self__._init_private_attributes()
 
-    @no_type_check
+    @no_type_check  # noqa: C901 (ignore complexity)
```

Comment:
```shell
$ git commit                             
Check Yaml...........................................(no files to check)Skipped
Fix End of Files.........................................................Passed
Lint.....................................................................Failed
- hook id: lint
- exit code: 2

flake8 pydantic/ tests/
pydantic/main.py:340:5: C901 'BaseModel.__setattr__' is too complex (16)
make: *** [lint] Error 1
```
The `no_type_check` decorator seems to need `noqa` in Python 3.7.
If you know another solution, please tell me.